Importer batch upserter fails

### Description

Importer batch upserter fails in integration envrionment, logs as follow:

```
2023-07-31T15:37:09.333-0600 ERROR scheduling-6 c.h.m.i.p.r.RecordFileParser Error parsing file 2023-07-31T21_27_44.008422618Z.rcd.gz after 95.92 ms com.hedera.mirror.importer.exception.ParserException: Error copying 10 items to table entity_temp
	at com.hedera.mirror.importer.parser.batch.BatchInserter.persist(BatchInserter.java:115)
	at com.hedera.mirror.importer.parser.batch.CompositeBatchPersister.persist(CompositeBatchPersister.java:72)
	at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.flush(SqlEntityListener.java:559)
	at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:227)
	at com.hedera.mirror.importer.parser.record.entity.sql.SqlEntityListener.onEnd(SqlEntityListener.java:90)
	at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:44)
	at com.hedera.mirror.importer.parser.record.entity.CompositeRecordStreamFileListener.onEnd(CompositeRecordStreamFileListener.java:29)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:145)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.doParse(RecordFileParser.java:47)
	at com.hedera.mirror.importer.parser.AbstractStreamFileParser.parse(AbstractStreamFileParser.java:79)
	at com.hedera.mirror.importer.parser.record.RecordFileParser.parse(RecordFileParser.java:118)
	at jdk.internal.reflect.GeneratedMethodAccessor138.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:343)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:391)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.retry.interceptor.RetryOperationsInterceptor$1.doWithRetry(RetryOperationsInterceptor.java:102)
	at org.springframework.retry.support.RetryTemplate.doExecute(RetryTemplate.java:329)
	at org.springframework.retry.support.RetryTemplate.execute(RetryTemplate.java:209)
	at org.springframework.retry.interceptor.RetryOperationsInterceptor.invoke(RetryOperationsInterceptor.java:135)
	at org.springframework.retry.annotation.AnnotationAwareRetryOperationsInterceptor.invoke(AnnotationAwareRetryOperationsInterceptor.java:160)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:756)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.hedera.mirror.importer.parser.record.RecordFileParser$$SpringCGLIB$$0.parse(<generated>)
	at com.hedera.mirror.importer.config.MessagingConfiguration.lambda$integrationFlow$1(MessagingConfiguration.java:107)
	at org.springframework.integration.handler.LambdaMessageProcessor.invokeMethod(LambdaMessageProcessor.java:208)
	at org.springframework.integration.handler.LambdaMessageProcessor.processMessage(LambdaMessageProcessor.java:121)
	at org.springframework.integration.handler.ServiceActivatingHandler.handleRequestMessage(ServiceActivatingHandler.java:95)
	at org.springframework.integration.handler.AbstractReplyProducingMessageHandler.handleMessageInternal(AbstractReplyProducingMessageHandler.java:136)
	at org.springframework.integration.handler.AbstractMessageHandler.doHandleMessage(AbstractMessageHandler.java:105)
	at org.springframework.integration.handler.AbstractMessageHandler.handleWithMetrics(AbstractMessageHandler.java:90)
	at org.springframework.integration.handler.AbstractMessageHandler.handleMessage(AbstractMessageHandler.java:70)
	at org.springframework.integration.endpoint.PollingConsumer.handleMessage(PollingConsumer.java:158)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.messageReceived(AbstractPollingEndpoint.java:474)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.doPoll(AbstractPollingEndpoint.java:460)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.pollForMessage(AbstractPollingEndpoint.java:412)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$4(AbstractPollingEndpoint.java:348)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.lambda$execute$0(ErrorHandlingTaskExecutor.java:57)
	at org.springframework.core.task.SyncTaskExecutor.execute(SyncTaskExecutor.java:50)
	at org.springframework.integration.util.ErrorHandlingTaskExecutor.execute(ErrorHandlingTaskExecutor.java:55)
	at org.springframework.integration.endpoint.AbstractPollingEndpoint.lambda$createPoller$5(AbstractPollingEndpoint.java:341)
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54)
	at org.springframework.scheduling.concurrent.ReschedulingRunnable.run(ReschedulingRunnable.java:96)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.hedera.mirror.importer.exception.ParserException: Error copying 10 items to table entity
	at com.hedera.mirror.importer.parser.batch.BatchUpserter.persistItems(BatchUpserter.java:84)
	at com.hedera.mirror.importer.parser.batch.BatchInserter.persist(BatchInserter.java:111)
	... 56 more
Caused by: org.postgresql.util.PSQLException: ERROR: invalid value for parameter "temp_buffers": 32768
  Detail: "temp_buffers" cannot be changed after any temporary tables have been accessed in the session.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2713)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2401)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:368)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:498)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:415)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:177)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at com.hedera.mirror.importer.parser.batch.BatchUpserter.createTempTable(BatchUpserter.java:90)
at com.hedera.mirror.importer.parser.batch.BatchUpserter.persistItems(BatchUpserter.java:75)
	... 57 more
```

### Steps to reproduce

see the logs above

### Additional context

_No response_

### Hedera network

other

### Version

v0.86.0-SNAPSHOT

### Operating system

None

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/CommonParserProperties.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/CommonParserProperties.java
@@ -46,7 +46,7 @@ public class CommonParserProperties {
     private Collection<TransactionFilter> include = new ArrayList<>();
 
     @Min(1)
-    private int tempTableBufferSize = 256; // Size in MB
+    private int tempTableBufferSize; // Default value set in application.yml, size in MB
 
     @Getter(lazy = true)
     private final Predicate<TransactionFilterFields> filter = includeFilter().and(excludeFilter());

--- a/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/batch/BatchUpserter.java
+++ b/hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/batch/BatchUpserter.java
@@ -49,12 +49,10 @@ public class BatchUpserter extends BatchInserter {
             CommonParserProperties properties,
             UpsertQueryGenerator upsertQueryGenerator) {
         super(entityClass, dataSource, meterRegistry, properties, upsertQueryGenerator.getTemporaryTableName());
-        String createTempIndexSql = upsertQueryGenerator.getCreateTempIndexQuery();
-        String createTempTableSql = upsertQueryGenerator.getCreateTempTableQuery();
-        String setTempBuffersSql = String.format("set temp_buffers = '%dMB'", properties.getTempTableBufferSize());
-        String truncateSql = String.format("truncate table %s restart identity cascade", tableName);
-        tempTableSql =
-                StringUtils.joinWith(";\n", setTempBuffersSql, createTempTableSql, createTempIndexSql, truncateSql);
+        var createTempIndexSql = upsertQueryGenerator.getCreateTempIndexQuery();
+        var createTempTableSql = upsertQueryGenerator.getCreateTempTableQuery();
+        var truncateSql = String.format("truncate table %s restart identity cascade", tableName);
+        tempTableSql = StringUtils.joinWith(";\n", createTempTableSql, createTempIndexSql, truncateSql);
         finalTableName = upsertQueryGenerator.getFinalTableName();
         upsertSql = upsertQueryGenerator.getUpsertQuery();
         log.trace("Table: {}, Entity: {}, upsertSql:\n{}", finalTableName, entityClass, upsertSql);

--- a/hedera-mirror-importer/src/main/resources/application.yml
+++ b/hedera-mirror-importer/src/main/resources/application.yml
@@ -28,6 +28,8 @@ hedera:
           checksum: 2
         tokenAccountBalanceMigration:
           checksum: 3
+      parser:
+        tempTableBufferSize: 256 # Size in MB
 
 logging:
   level:
@@ -96,6 +98,7 @@ spring:
     url: jdbc:postgresql://${hedera.mirror.importer.db.host}:${hedera.mirror.importer.db.port}/${hedera.mirror.importer.db.name}?tcpKeepAlive=true
     username: ${hedera.mirror.importer.db.username}
     hikari:
+      connection-init-sql: set temp_buffers='${hedera.mirror.importer.parser.tempTableBufferSize}MB'
       # Note: Flyway does not use Hikari so these properties are ignored. Use URL properties for Flyway instead.
       data-source-properties:
         #loggerLevel: TRACE

